### PR TITLE
Show play overlay for video thumbnail placeholder

### PR DESCRIPTION
**Pull request by FeuRenard**
### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Sony Xperia U, Android 4.4.4 (CyanogenMod)
 * AVD Nexus 5X, Android 7.1
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description
Showing the play overlay above the video thumbnail placeholder should highlight, that the video can be played by clicking it.

### Screenshots
Android 4.4.4 KitKat
device-2017-05-13-152730
device-2017-05-13-152820
device-2017-05-13-155931

Android 7.1.1 Nougat
bildschirmfoto von 2017-05-13 15-44-33
bildschirmfoto von 2017-05-13 15-45-03

### CODE REVIEW

File: res/values/attrs.xml [hunk 1]
```diff
@@ -142,6 +142,7 @@
 
     <declare-styleable name="ThumbnailView">
         <attr name="backgroundColorHint" format="color" />
+        <attr name="thumbnail_play_overlay_tint" format="color" />
```
Comment (moxie0) [inline res/values/attrs.xml]: We usually define attributes this way if they are "owned" by the widget. For instance, if in specifying the widget via XML, we want to pass along some value to the code. This doesn't seem to be the case here from what I can tell, it's more of a theme thing right?

File: res/values/colors.xml [hunk 2]
```diff
@@ -55,4 +55,6 @@
 
     <color name="sticker_selected_color">#8cf437</color>
     <color name="transparent">#00FFFFFF</color>
+
+    <color name="thumbnail_play_overlay_tint">#77ffffff</color>
```
Comment (moxie0) [inline res/values/colors.xml]: Is there a reason this is a color but also an attribute that's defined differently in themes?

File: src/org/thoughtcrime/securesms/components/ThumbnailView.java [hunk 3]
```diff
@@ -107,9 +109,14 @@ public void setImageResource(@NonNull MasterSecret masterSecret, @NonNull Slide
       getTransferControls().setVisibility(View.GONE);
     }
 
-    if (slide.getThumbnailUri() != null && slide.hasPlayOverlay() &&
-        (slide.getTransferState() == AttachmentDatabase.TRANSFER_PROGRESS_DONE || isPreview))
-    {
+    if ((slide.getThumbnailUri() != null || slide.hasPlaceholder()) &&
+        slide.hasPlayOverlay()                                      &&
+        (slide.getTransferState() == AttachmentDatabase.TRANSFER_PROGRESS_DONE || isPreview)) {
+      if (slide.getThumbnailUri() != null) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/ThumbnailView.java]: Is the objective to only tint the placeholder according to theme, but to always tint the overlay to a thumbnail with the same color regardless of theme?

Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/components/ThumbnailView.java]: Exactly that.
White with the current tinting mode seems to be the best for thumbnail overlays. But the white one won't be visible over the dark theme placeholder.

File: res/values/colors.xml (hunk 2 above)
Comment (FeuRenard) [inline res/values/colors.xml]: See below.

File: res/values/attrs.xml (hunk 1 above)
Comment (FeuRenard) [inline res/values/attrs.xml]: You are right, this attribute is misplaced here. I will move it.

File: src/org/thoughtcrime/securesms/components/ThumbnailView.java (hunk 3 above)
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/ThumbnailView.java]: I think I get it, can we name those two things differently? It's kind of confusing to me that there's an attribute that's themed as well as a static color that's not themed, but they're both called the same thing.

Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/components/ThumbnailView.java]: I renamed the tint color for the placeholder (which changes with the theme) to `placeholder_play_overlay_tint`.

**Comment by FeuRenard** [comment]
I no longer need this PR for #6632 and I don't think this is very important. Moreover I consider the proposed overlay redundant information. We need the play overlay to distinguish between an image (no overlay) and a video thumbnail (overlay). But when there is a video placeholder image then we already know it's a video. So I see no longer a need for this overlay.

**Pull request closed without merging**